### PR TITLE
Preview builds -> download.delta.chat

### PR DESCRIPTION
**Pull request by lefherz**
Reworked the GitHub actions so they push the preview builds to https://download.delta.chat.

### CODE REVIEW

**Comment by lefherz** [comment]
Hm, I can't get the sed & the zip action to run right now. Pretty impossible to debug, because GitHub doesn't give me any logs or useful debug messages, only exit codes.

The sed action could probably just be skipped, although it makes testing less convenient. But without the zip action, we can't deliver the macOS & portable windows deliverable.

The zip action I'm using here is flawed, it doesn't even have an action.yml file. It definitely needs some work to get it running, currently it fails with exit code 14 or 15 or so. But it's pretty similar to the sed action I forked from it, so getting sed to run would be a good first step anyway.

@Simon-Laux what do you think? We could also only deliver .deb, .exe, and .AppImage for now.

**Comment by Simon-Laux** [comment]
I'll try somethings

**Comment by lefherz** [comment]
rssh blocks the scp action by appleboy, I think because it uses the scp -t flag. Log output of the delete action:

```Oct 31 11:42:57 page rssh[18734]: setting log facility to LOG_USER
Oct 31 11:42:57 page rssh[18734]: allowing scp to all users
Oct 31 11:42:57 page rssh[18734]: setting umask to 022
Oct 31 11:42:57 page rssh[18734]: cmd 'scp' approved
Oct 31 11:42:57 page rssh[18734]: cmd 'scp' approved
Oct 31 11:42:57 page rssh[18734]: insecure scp option in scp command line
Oct 31 11:42:57 page rssh[18734]: user jekyll attempted to execute forbidden commands
Oct 31 11:42:57 page rssh[18734]: command: scp -tr isE3Yjl3Iy.tar
```

File: .github/workflows/build-preview.yml [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 jobs:
   build:
 
     runs-on: ubuntu-latest
 
     steps:
+    - name: Get Pullrequest ID
+      id: prepare
+      run: |
+        export PULLREQUEST_ID=$(echo "refs/pull/1088/merge" | cut -d '/' -f3)
```
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply .github/workflows/build-preview.yml]: right, `"refs/pull/1088/merge"` should be replaced by `"${{ github.ref }}"`

File: .github/workflows/build-preview.yml [hunk 2]
```diff
@@ -1,13 +1,24 @@
+# documentation: https://github.com/deltachat/sysadmin/tree/master/download.delta.chat
 name: Preview
 
-on: [pull_request]
+on: 
+  pull_request:
+    paths-ignore:
+      - 'docs/**'  # only trigger build if a file outside of /docs was changes
```
Comment (Jikstra) [inline .github/workflows/build-preview.yml]: `changes` -> `changed`

Comment (lefherz) [reply .github/workflows/build-preview.yml]: Thanks, missed it.

**Comment by lefherz** [comment]
Hm, the delete doesn't seem to work - github.ref was apparently `master` at the close event. So now we have the following files on page:

```
$ ls -l /var/www/html/download/desktop/preview/
total 380244
-rwxr-xr-x 1 jekyll jekyll 109980004 Nov  1 15:28 deltachat-desktop-1088.AppImage
-rw-r--r-- 1 jekyll jekyll  71629392 Nov  1 15:28 deltachat-desktop-1088.exe
-rw-r--r-- 1 jekyll jekyll 207744451 Nov  1 15:28 deltachat-desktop-mas-1088.zip
-rw-r--r-- 1 jekyll jekyll        53 Nov  1 17:01 deltachat-desktop-mas-master.zip
-rw-r--r-- 1 jekyll jekyll        53 Nov  1 17:01 deltachat-desktop-master.AppImage
-rw-r--r-- 1 jekyll jekyll        53 Nov  1 17:01 deltachat-desktop-master.exe
```

**Pull request merged**